atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTableTest.java

```diff
@@ -0,0 +1,151 @@
+        SqlConnection sqlConnection = setUpSqlConnection(null, new Timestamp(95), null, new Timestamp(92));
+        assertThatVacuumWasNotPerformed(sqlConnection);
+    }
+
+
+    @Test
+    public void shouldNotCompactIfVacuumWasPerformedWithinCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(new Timestamp(95), null, new Timestamp(92), null);
+        assertThatVacuumWasNotPerformed(sqlConnection);
+    }
+
+    @Test
+    public void shouldNotCompactIfVacuumDidOccurInLastCompactMillis() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(new Timestamp(5), null, new Timestamp(90), new Timestamp(10));
+        assertThatVacuumWasNotPerformed(sqlConnection);
+    }
+
+    private SqlConnection setUpSqlConnection(Timestamp lastVacuumTimestamp, Timestamp lastAutoVacuumTimestamp,
+            Timestamp lastAnalyzeTimestamp, Timestamp lastAutoAnalyzeTimestamp) {
+        SqlConnection sqlConnection = mock(SqlConnection.class);
+        when(connectionSupplier.get()).thenReturn(sqlConnection);
+
+        List<List<Object>> selectResults = new ArrayList<>();
+        selectResults.add(Arrays.asList(
+                new Object[] {"ns__test", lastVacuumTimestamp, lastAutoVacuumTimestamp, lastAnalyzeTimestamp,
+                              lastAutoAnalyzeTimestamp}));
+
+        List<List<Object>> timestampList = new ArrayList<>();
+        timestampList.add(Arrays.asList(new Object[] {new Timestamp(NOW_MILLIS)}));
+
+        when(sqlConnection.selectResultSetUnregisteredQuery(eq("SELECT CURRENT_TIMESTAMP")))
+                .thenReturn(new AgnosticResultSetImpl(timestampList,
+                        DBType.POSTGRESQL,
+                        new ImmutableMap.Builder<String, Integer>()
+                                .put("now", 0)
+                                .put("NOW", 0)
+                                .build()));
+
+        Mockito.when(sqlConnection.selectResultSetUnregisteredQuery(startsWith("SELECT relname"), any()))
+                .thenReturn(
+                        new AgnosticResultSetImpl(selectResults,
+                                DBType.POSTGRESQL,
+                                new ImmutableMap.Builder<String, Integer>()
+                                        .put("relname", 0)
+                                        .put("RELNAME", 0)
+                                        .put("LAST_AUTOVACUUM", 2)
+                                        .put("last_autovacuum", 2)
+                                        .put("last_analyze", 3)
+                                        .put("LAST_ANALYZE", 3)
+                                        .put("last_vacuum", 1)
+                                        .put("LAST_VACUUM", 1)
+                                        .put("last_autoanalyze", 4)
+                                        .put("LAST_AUTOANALYZE", 4)
+                                        .build()));
```

Comment:
We should probably document (in a comment, maybe) why we're returning both lower and upper case versions of the strings.

Also nit: extract a method `putBothLowerAndUpperCase(ImmutableMap.Builder builder, String string, int value)`?

Reply:
Added a comment, dont think having a method will make it any cleaner, so skipping.